test/e2e/testsuites/istio.go

```diff
@@ -0,0 +1,135 @@
+func InitGcsFuseCSIIstioTestSuite() storageframework.TestSuite {
+	return &gcsFuseCSIIstioTestSuite{
+		tsInfo: storageframework.TestSuiteInfo{
+			Name: "volumes",
+			TestPatterns: []storageframework.TestPattern{
+				storageframework.DefaultFsCSIEphemeralVolume,
+				storageframework.DefaultFsPreprovisionedPV,
+				storageframework.DefaultFsDynamicPV,
+			},
+		},
+	}
+}
+
+func (t *gcsFuseCSIIstioTestSuite) GetTestSuiteInfo() storageframework.TestSuiteInfo {
+	return t.tsInfo
+}
+
+func (t *gcsFuseCSIIstioTestSuite) SkipUnsupportedTests(_ storageframework.TestDriver, _ storageframework.TestPattern) {
+}
+
+func (t *gcsFuseCSIIstioTestSuite) DefineTests(driver storageframework.TestDriver, pattern storageframework.TestPattern) {
+	envVar := os.Getenv(utils.TestWithNativeSidecarEnvVar)
+	gcsFuseSupportsNativeSidecar, err := strconv.ParseBool(envVar)
+	if err != nil {
+		klog.Fatalf(`env variable "%s" could not be converted to boolean`, envVar)
+	}
+
+	type local struct {
+		config         *storageframework.PerTestConfig
+		volumeResource *storageframework.VolumeResource
+	}
+	var l local
+	ctx := context.Background()
+
+	f := framework.NewFrameworkWithCustomTimeouts("istio", storageframework.GetDriverTimeouts(driver))
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	init := func(configPrefix ...string) {
+		l = local{}
+		l.config = driver.PrepareTest(ctx, f)
+		if len(configPrefix) > 0 {
+			l.config.Prefix = configPrefix[0]
+		}
+		l.volumeResource = storageframework.CreateVolumeResource(ctx, driver, l.config, pattern, e2evolume.SizeRange{})
+	}
+
+	cleanup := func() {
+		var cleanUpErrs []error
+		cleanUpErrs = append(cleanUpErrs, l.volumeResource.CleanupResource(ctx))
+		err := utilerrors.NewAggregate(cleanUpErrs)
+		framework.ExpectNoError(err, "while cleaning up")
+	}
+
+	testSidecarStoreDataSupportScenario := func(setGcsFuseNativeSidecar bool, setIstioNativeSidecar bool) {
+		init()
+		defer cleanup()
+
+		ginkgo.By("Configuring the pod")
+		tPod := specs.NewTestPod(f.ClientSet, f.Namespace)
+		tPod.SetupVolume(l.volumeResource, volumeName, mountPath, false)
+		tPod.SetIstioSidecar(setIstioNativeSidecar)
+
+		ginkgo.By("Deploying the pod")
+		tPod.Create(ctx)
+		defer tPod.Cleanup(ctx)
+
+		ginkgo.By("Checking that the pod is running")
+		tPod.WaitForRunning(ctx)
+
+		ginkgo.By("Checking injection ordering")
+		tPod.VerifyInjectionOrder(setGcsFuseNativeSidecar, setIstioNativeSidecar)
+
+		ginkgo.By("Checking that the pod command exits with no error")
+		tPod.VerifyExecInPodSucceed(f, specs.TesterContainerName, fmt.Sprintf("mount | grep %v | grep rw,", mountPath))
+		tPod.VerifyExecInPodSucceed(f, specs.TesterContainerName, fmt.Sprintf("echo 'hello world' > %v/data && grep 'hello world' %v/data", mountPath, mountPath))
+	}
+
+	// We do not support the case where:
+	// - GCSFuse is native sidecar.
+	// - Istio is a regular container.
+	ginkgo.It("should store data with istio regular container present", func() {
```

Comment:
remove "present" to keep it consistent, or add "present" to the next test case name